x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -429,6 +432,13 @@ export function App({
         });
         redirectToOrigin({ input: newInput, isCopied: saveProps.newCopyOnSave });
         return;
+      } else if (saveProps.dashboardId && redirectToDashboard) {
+        // disabling the validation on app leave because the document has been saved.
+        onAppLeave((actions) => {
+          return actions.default();
+        });
+        redirectToDashboard(newInput, saveProps.dashboardId);
```

Comment:
Wondering if we should insert a listing page history item in here like Visualize does in order to align the behaviour when pressing the back button after one of these redirects.